app/javascript/app/detail/detail-map.js

```diff
@@ -0,0 +1,92 @@
+// Manages location detail view Google Map.
+import markers from 'app/util/map/markers.js.erb';
+import googleMaps from 'app/util/map/google/map-loader';
+
+function init() {
+  googleMaps.load(_renderMap);
+}
+
+// @param nodes [Array] List of DOM HTMLNodes.
+function _removeDOMNodeList( nodes ) {
+  var node;
+  for ( var i = 0, len = nodes.length; i < len; i++ ) {
+    node = nodes[i];
+    node.parentNode.removeChild( node );
+  }
+}
+
+function _renderMap() {
+  var mapCanvas = document.getElementById('detail-map-canvas');
+
+  if (mapCanvas) {
+    var infoWindow = new google.maps.InfoWindow();
+    var titleElm = document.getElementById('detail-map-canvas-title');
+    var latElm = document.getElementById('detail-map-canvas-lat');
+    var lngElm = document.getElementById('detail-map-canvas-lng');
+
+    // Retrieve marker data from embedded HTML values.
+    var title = titleElm.innerHTML.trim();
+    var lat = latElm.innerHTML.trim();
+    var lng = lngElm.innerHTML.trim();
+
+    // Remove embedded values from the DOM.
+    _removeDOMNodeList( [titleElm, latElm, lngElm] );
+
+    var latLng = new google.maps.LatLng(lat, lng);
+
+    // Turns off Google Points-Of-Interest (POI) markers so the user
+    // doesn't click a POI and get an infowindow popped up.
+    var poiStyles =[
+      {
+        featureType: 'poi',
+        elementType: 'labels',
+        stylers: [
+          { visibility: 'off' }
+        ]
+      }
+    ];
+
+    var mapOptions = {
+      maxZoom: 16,
+      center: latLng,
+      scrollwheel: false,
+      zoomControl: true,
+      panControl: false,
+      streetViewControl: false,
+      scaleControl: true,
+      scaleControlOptions: {
+        position: google.maps.ControlPosition.RIGHT_BOTTOM
+      },
+
```

Comment:
We may want to add `fullscreenControl: false,` to prevent the fullscreen button from appearing on the map. It doesn't work well with the popup when the map is expanded.

![Uploading Screen Shot 2018-03-02 at 10.49.16 PM.png…]()

Reply:
Which popup are you referring to? The info box when you click on one of the markers? I tried it in Chrome and Safari in fullscreen mode and it seems to work fine, at least not any worse than when in regular screen mode.

Reply:
That's true, I guess I just notice it more at full screen size since there's more room for the markers to zoom off screen. Like if you go to the top-most one and then the bottom-most. Map could use some adjustments 🙁